fcrepo-kernel/src/main/java/org/fcrepo/kernel/exception/ConstraintViolationException.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.exception;
+
+import javax.ws.rs.core.UriInfo;
```

Comment:
We do not want a `javax.ws.rs` dependency in `fcrepo-kernel`. You should be able to remove the following method:
https://github.com/fcrepo4/fcrepo4/pull/808/files#diff-c14ecfb343036db02d3070d59bd47b0eR64
...and perform all of the `buildConstraintLink` logic in `ConstraintExceptionMapper`, no?